src/base58.cpp

```diff
@@ -136,7 +136,7 @@ bool DecodeBase58Check(const char* psz, std::vector<unsigned char>& vchRet)
     }
     // re-calculate the checksum, ensure it matches the included 4-byte checksum
     uint256 hash = Hash(vchRet.begin(), vchRet.end() - 4);
-    if (memcmp(&hash, &vchRet.end()[-4], 4) != 0) {
+    if (memcmp(&hash, vchRet.data() + vchRet.size() - 4, 4) != 0) {
```

Comment:
I guess there's some reason `vchRet.end() - 4` doesn't work (pointer versus iterator conflict or something like that)?